### PR TITLE
refine logs print and make e2e test configurable

### LINKED ISSUE #206: Worker Timeout causing query job PANIC
according to the log, WorkerInfo is nil but not set tombstone,

log
```
[2022/03/08 00:14:26.968 +08:00] [WARN] [worker_manager.go:548] ["Worker timed out"] [duration=15.071926185s] [id=81629056-8a99-4a61-8821-6b7cf22ad908]
[2022/03/08 00:14:26.968 +08:00] [DEBUG] [worker_manager.go:287] ["Sending heartbeat response to worker"] [worker-id=2736eaea-6a24-4736-875f-5f46d1e9708b] [worker-node-id=11195c75-92e5-438d-a08f-315f17c4f319] [message="{\"send-time\":1474070339601315,\"reply-time\":\"2022-03-08T00:14:26.968473503+08:00\",\"to-worker-id\":\"2736eaea-6a24-4736-875f-5f46d1e9708b\",\"epoch\":9}"]
[2022/03/08 00:14:26.968 +08:00] [DEBUG] [worker_manager.go:287] ["Sending heartbeat response to worker"] [worker-id=18307cbb-0c49-42f6-93c0-24706cdff053] [worker-node-id=48b3bfcf-5ca0-4484-ab5f-7c16d61f9c49] [message="{\"send-time\":1474391440873062,\"reply-time\":\"2022-03-08T00:14:26.968654875+08:00\",\"to-worker-id\":\"18307cbb-0c49-42f6-93c0-24706cdff053\",\"epoch\":9}"]
[2022/03/08 00:14:26.968 +08:00] [WARN] [worker_manager.go:548] ["Worker timed out"] [duration=15.273693124s] [id=09c2221d-7ebe-4eb3-b7e0-3304777a943f]
[2022/03/08 00:14:26.968 +08:00] [INFO] [master.go:360] ["worker is offline"] [master-id=dataflow-engine-job-manager] [worker-info="{\"ID\":\"81629056-8a99-4a61-8821-6b7cf22ad908\",\"NodeID\":\"7ccdcf7f-e4f1-46ce-b05b-51c31a3a8e3e\"}"]
[2022/03/08 00:14:26.969 +08:00] [DEBUG] [interceptor.go:181] ["request stats"] [component="embed etcd"] ["start time"=2022/03/08 00:14:26.969 +08:00] ["time spent"=46.382µs] [remote=10.244.6.55:50444] ["response type"=/pb.Master/QueryJob] ["request cc
ount"=-1] ["request size"=-1] ["response count"=-1] ["response size"=-1] ["request content"=]
[2022/03/08 00:14:26.970 +08:00] [PANIC] [job_fsm.go:124] ["Unexpected Job Info"] [stack="github.com/hanfei1991/microcosm/servermaster.(*JobFsm).QueryJob\n\t/root/microcosm/servermaster/job_fsm.go:124\ngithub.com/hanfei1991/microcosm/servermaster.(*JobManagerImplV2).QueryJob\n\t/root/microcosm/servermaster/jobmanager.go:56\ngithub.com/hanfei1991/microcosm/servermaster.(*Server).QueryJob\n\t/root/microcosm/servermaster/server.go:221\ngithub.com/hanfei1991/microcosm/pb._Master_QueryJob_Handler.func1\n\t/root/microcosm/pb/master.pb.go:1576\ngithub.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1\n\t/root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n\t/root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.1.0/chain.go:25\ngo.etcd.io/etcd/etcdserver/api/v3rpc.newUnaryInterceptor.func1\n\t/root/go/pkg/mod/go.etcd.io/etcd@v0.5.0-alpha.5.0.20210512015243-d19fbe541bf9/etcdserver/api/v3rpc/interceptor.go:70\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n\t/root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.1.0/chain.go:25\ngo.etcd.io/etcd/etcdserver/api/v3rpc.newLogUnaryInterceptor.func1\n\t/root/go/pkg/mod/go.etcd.io/etcd@v0.5.0-alpha.5.0.20210512015243-d19fbe541bf9/etcdserver/api/v3rpc/interceptor.go:77\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n\t/root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.1.0/chain.go:25\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1\n\t/root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.1.0/chain.go:34\ngithub.com/hanfei1991/microcosm/pb._Master_QueryJob_Handler\n\t/root/microcosm/pb/master.pb.go:1578\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/root/go/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1082\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/root/go/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1405\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.1\n\t/root/go/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:746"]
~
```

### LINKED ISSUE #209: Schedule result is not even
As observation, most jobmasters are located in same node (cos2):
```
cos2  submit job 930fd67c-bc14-4bf3-93e7-a81e61a36bd7
cos2  submit job 2c6899db-0e51-448e-aa39-bc41d939c2ac
cos3  submit job b47e9290-7fb5-4e67-946a-9395d65c765d
cos1  submit job 389bc3eb-7e75-4294-b7e6-93611fa7ba41
cos2  submit job 1f1a2a08-006d-455c-884e-8b9f244be7e3
cos2  submit job d42aa555-22c2-4050-9ac0-8e6e29575ced
cos2  submit job 09a37f65-81a9-4d48-b9d1-ee5b97c138ca
cos3  submit job abddc49c-b685-4c41-a47d-5a6feefc06db
cos3  submit job de3b6ac8-24eb-4fec-9dde-ee5b78262dc5
cos2  submit job 49998e05-cc9d-48ba-9e5f-8f0c975d344e
cos2  submit job 5e95013e-aad5-4b8b-a4a4-f8a438178449
cos2  submit job 16430e90-4862-4269-8035-bbef00f52244
cos2  submit job c4c1b656-edb5-4963-8b42-ee9651621c24
cos2  submit job 107cb3a1-0f58-4d78-8ca4-bf51c378ffac
cos2  submit job 245bc0eb-82da-4010-8730-15abffd7aa26
```

### PR DESCRIPTION
**Pull request by hanfei1991**
1. add more log details
2. fix #209, use random int.
3. fix #206 

Fix query job bugs about PANIC. When jobmaster timeout, the base master will delete its worker handler, at the same time, fsm never stores tombstone handler, then cause PANIC.

### CODE REVIEW

File: jobmaster/cvsJob/cvsJobMaster.go [hunk 1]
```diff
@@ -95,13 +95,20 @@ func (jm *JobMaster) InitImpl(ctx context.Context) error {
...
 	}
-	jm.status = lib.WorkerStatusNormal
-	return jm.UpdateJobStatus(ctx, jm.Status())
+	return nil
 }
 
 func (jm *JobMaster) Tick(ctx context.Context) error {
 	filesNum := 0
 	jm.counter = 0
+	if jm.status != lib.WorkerStatusNormal {
```
Comment (nongfushanquan) [inline jobmaster/cvsJob/cvsJobMaster.go:104]: if  ！= workerstatusnormal  ,should return error at once

Comment (hanfei1991) [reply jobmaster/cvsJob/cvsJobMaster.go:104]: if != normal, it's supposed to be init. I set it in "Init" function at first, but meet `update status` retry fail multiple times. So I retry `update status` in Tick function

File: servermaster/jobmanager.go [hunk 2]
```diff
@@ -69,6 +69,9 @@ func (jm *JobManagerImplV2) QueryJob(ctx context.Context, req *pb.QueryJobReques
 			}
 			return resp
 		}
+		if masterMeta != nil {
+			log.L().Warn("load master kv meta from meta store, but status is not expected", zap.Any("id", req.JobId), zap.Any("status", masterMeta.StatusCode))
+		}
```
Comment (amyangfei) [inline servermaster/jobmanager.go:74]: How to reproduce this unepxected case in test?

File: servermaster/resource/capacity_impl.go [hunk 3]
```diff
@@ -87,7 +89,8 @@ func (m *CapRescMgr) allocateTasksWithNaiveStrategy(
 		// No resources in this cluster
 		return false, nil
 	}
-	var idx int = 0
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	idx := r.Intn(len(resources))
```
Comment (amyangfei) [inline servermaster/resource/capacity_impl.go:93]: What about storing a `*rand.Rand` field in `CapRescMgr`, then we don't need to create a new seed every time.

File: servermaster/jobmanager.go (hunk 2 above)
Comment (hanfei1991) [reply servermaster/jobmanager.go:74]: After thinking twice, It seems this branch is impossible. If jobmaster offline normally, the meta should be set finished, or it needs failover and is in pending queue.

Comment (hanfei1991) [reply servermaster/jobmanager.go:74]: How about keeping this log as an unharmful warn?

Comment (amyangfei) [reply servermaster/jobmanager.go:74]: Fine

**Review by amyangfei — APPROVED** [review]

**Pull request merged**